examples/run/cuda/seq_example_cuda.cpp

```diff
@@ -0,0 +1,282 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+
+// vecmem
+#include <vecmem/memory/cuda/managed_memory_resource.hpp>
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// algorithms
+#include "clusterization/clusterization_algorithm.hpp"
+#include "cuda/track_finding/seeding_algorithm.hpp"
+#include "track_finding/seeding_algorithm.hpp"
+
+// io
+#include "io/csv.hpp"
+#include "io/reader.hpp"
+
+int seq_run(const std::string& detector_file, const std::string& cells_dir,
+            unsigned int skip_events, unsigned int events, bool skip_cpu,
+            bool skip_write) {
+    auto env_d_d = std::getenv("TRACCC_TEST_DATA_DIR");
+    if (env_d_d == nullptr) {
+        throw std::ios_base::failure(
+            "Test data directory not found. Please set TRACCC_TEST_DATA_DIR.");
+    }
+    auto data_directory = std::string(env_d_d) + std::string("/");
+
+    // Read the surface transforms
+    auto surface_transforms = traccc::read_geometry(detector_file);
+
+    // Output stats
+    uint64_t n_cells = 0;
+    uint64_t n_modules = 0;
+    uint64_t n_clusters = 0;
+    uint64_t n_measurements = 0;
+    uint64_t n_spacepoints = 0;
+    uint64_t n_internal_spacepoints = 0;
+    uint64_t n_seeds = 0;
+    uint64_t n_seeds_cuda = 0;
+
+    // Elapsed time
+    float wall_time(0);
+    float file_reading_cpu(0);
+    float clusterization_cpu(0);
+    float seeding_cpu(0);
+    float clusterization_cuda(0);
+    float seeding_cuda(0);
+
+    // Memory resource used by the EDM.
+    vecmem::host_memory_resource host_mr;
+
+    // Memory resource used by the EDM.
+    vecmem::cuda::managed_memory_resource mng_mr;
+
+    traccc::clusterization_algorithm ca;
+    traccc::cuda::seeding_algorithm sa_cuda(&mng_mr);
+    traccc::seeding_algorithm sa(&host_mr);
+
+    /*time*/ auto start_wall_time = std::chrono::system_clock::now();
+
+    // Loop over events
+    for (unsigned int event = skip_events; event < skip_events + events;
+         ++event) {
+
+        // Read the cells from the relevant event file
+
+        /*-----------------------------
+              Read the cell data
+          -----------------------------*/
+
+        /*time*/ auto start_file_reading_cpu = std::chrono::system_clock::now();
+
+        std::string event_string = "000000000";
```

Comment:
There are some utils regarding reading in [reader.hpp](https://github.com/acts-project/traccc/blob/556542b39feef4b3cda06d44f2becb6adc7fda0f/io/include/io/reader.hpp#L26)

Reply:
Fixed it